e2e/jail/jail_rpc_test.go

```diff
@@ -33,6 +34,13 @@ func (s *JailRPCTestSuite) SetupTest() {
 	s.BackendTestSuite.SetupTest()
 	s.jail = s.Backend.JailManager()
 	s.NotNil(s.jail)
+
+	s.Do(func() {
+		// Testnet sync
+		s.StartTestBackend()
+		EnsureNodeSync(s.Backend.NodeManager())
+		s.StopTestBackend()
+	})
```

Comment:
We've normally done it in `SetupSuite`, is there a reason not to do so this time?

Reply:
Yep, i need all initialization of `SetupTest` to run this methods. Just dont want make it twice.

Reply:
Wait. There's code in many tests which actually require blockchain sync, it's not in `SetupSuite`:
```
	s.StartTestBackend()
	defer s.StopTestBackend()

	EnsureNodeSync(s.Backend.NodeManager())
```

Why have you added it in `SetupTest`?

Reply:
To make sync before test. If test depends on synced state we should prepate this state in SetupTest or SetupSuite. I've just extract all this logic into SetupTest and TeardownTest.

Reply:
Still in the dark.

There're tests which require a started node and synced state. Those tests start a node, ensure it's synced and stop the node.
Those tests which don't require a synced node, don't do that.
This approach has the following advantages and disadvantages:
+ it's clear if the test requires a started node from reading the test code
+ only those tests which require a synced blockchain do start-sync-stop round, which keep amount of time for running tests minimal
- you have to type these 3 lines of code for each test requiring a started node

Put putting them in `SetupTest` means that it:
+ frees us from writing those 3 lines in each test requiring synced blockchain
- makes it unclear from reading the code which tests require it and which don't
- makes tests run longer due to those start-sync-stop cycles

Even if you put this code into `SetupSuite`, you'll only turn one - into + still leaving unclarity about which tests require synced blockchain and which don't. And I find the latter the most important and believe it's worth 3 lines of code in each test.

Reply:
I think it is unclear even now: a part of init stage is in tests themself, if one test done sync than all the following tests is in sync or nearly sync state and we cant say that test that doesnt do `EnsureNodeSync(s.Backend.NodeManager())` is in unsync state (actually it is in synked state because first test `TestJailRPCSend` do the sync and doesnt clear chain in Teardown so all other tests run in synced state - they all do `s.StartTestBackend()`).
If we want 2 types of tests i see such possobilities:
1. Add new chain in new directory
2. Make 2 test suetes: synced and unsynced suites and guarantee that the first one should run as first. Hacky way but better than cuttent tests.

Reply:
Yes, that's the idea: if a test needs a synced blockchain, it does it for the sake of itself and other future tests, it's by design. You also ignored all my arguments.
What problem are you solving?